### PR TITLE
Add an abstract delivery_detail class

### PR DESCRIPTION
**Pull request by vkrmis**
**Story card:** [sc-7663](https://app.shortcut.com/simpledotorg/story/7663/abstract-delivery-details)

## Because

We need to handle delivery details on a per SMS service provider basis since they can differ significantly

## This addresses

Adds an abstract `DeliveryDetail` class

### CODE REVIEW

File: app/models/twilio_sms_delivery_detail.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-class TwilioSmsDeliveryDetail < ApplicationRecord
+class TwilioSmsDeliveryDetail < DeliveryDetail
   has_one :communication, as: :detailable
```
Comment (harimohanraj89) [inline app/models/twilio_sms_delivery_detail.rb:2]: Does this go in the abstract class?

**Review by harimohanraj89 — APPROVED** [review]
One question - non blocking.

Looks like we've already set up a polymorphic `detailable` association to the Communication, eh? Makes this a nice continuation of that thought, without having to update the touchpoints where we access the detailable. Nice.

**Comment by vkrmis** [comment]
> One question - non blocking.
> 
> Looks like we've already set up a polymorphic `detailable` association to the Communication, eh? Makes this a nice continuation of that thought, without having to update the touchpoints where we access the detailable. Nice.

Yup, pretty much. Also allows each delivery detail type to define their own criteria for success, failure, in_progress.

File: app/models/twilio_sms_delivery_detail.rb (hunk 1 above)
Comment (vkrmis) [reply app/models/twilio_sms_delivery_detail.rb:2]: moved 🙏

**Pull request merged**